### PR TITLE
DRILL-6054: don't try to split the filter when it is not AND

### PR DESCRIPTION
**Pull request by chunhui-shi**

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/FindPartitionConditions.java [hunk 1]
```diff
@@ -228,13 +236,16 @@ private boolean isHolisticExpression(RexCall call) {
     return false;
   }
 
+  protected boolean inputRefToPush(RexInputRef inputRef) {
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/FindPartitionConditions.java]: Can be made private. Should be placed under public methods.

File: exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/FindPartitionConditions.java [hunk 2]
```diff
@@ -195,8 +195,16 @@ private void popOpStackAndBuildFilter() {
          * For all other operators we clear the children if one of the
          * children is a no push.
          */
-        assert currentOp.getOp().getKind() == SqlKind.AND;
-        newFilter = currentOp.getChildren().get(0);
+        if (currentOp.getOp().getKind() == SqlKind.AND) {
+          newFilter = currentOp.getChildren().get(0);
+          for(OpState opState : opStack) {
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/FindPartitionConditions.java]: Please add space: `for (`

File: exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/FindPartitionConditions.java (hunk 1 above)
Comment (chunhui-shi) [reply exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/FindPartitionConditions.java]: This is intentionally made to be 'protected' for future extension.
Right now, FindPartitionCondition use position based inputRef(using BitSet dirs) to mark which inputRef should be pushed. But in future, we may use name based policy to decide which one to push.

File: exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/FindPartitionConditions.java (hunk 2 above)
Comment (chunhui-shi) [reply exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/FindPartitionConditions.java]: done.

**Comment by arina-ielchiieva** [comment]
+1, LGTM.

**Pull request closed without merging**